### PR TITLE
fix: Adapt content script to the new mozsearch SYM_INFO format

### LINKED ISSUE #10: Searchfox ANALYSIS_DATA went away and probably broke searchfox integration
In https://bugzilla.mozilla.org/show_bug.cgi?id=1776522 I'm planning to move searchfox from using ANALYSIS_DATA keyed by data-i to SYM_INFO (or successor) keyed by the content of data-symbols split on `,`.  This should be up on dev.searchfox.org within a few days at which point the revised SYM_INFO schema should be locked down, but I'm unable to provide a precise estimate for when the change will happen on searchfox.org.

### PR DESCRIPTION
**Pull request by rpl**
This PR contains a few small changes to the extension content script that are meant to adapt it to the new searchfox SYM_INFO format that has replaced the ANALYSIS_DATA (which the extension was using to generate the searchfox links into the phabricator review diff views).

Fixes #10

### CODE REVIEW

**Comment by rpl** [comment]
@marco-c I took a quick look into mozsearch sources and a bunch of pages on searchfox to get a picture of how the new format differs from the previous one (but I didn't have access to a dump of how the previous format looked like and so I went for guessing it based on how the phabricator.js content script was processing it).

I gave it a try on a some phabricator revision and the "jump to definition" and "jump to declaration" seems to be working now (and they should also impact a little less on searchfox, because the `jumps` in the new SYM_INFO format seems to be providing the direct path to include in the searchfox url and so searchfox will not have to re-run the same kind of query).

I also added a "Jump to IDL" link, because based on the mozsearch sources that may also be included in `jumps` for some symbols but none of the phabricator revision I've tried this PR on was including it and so I was not able to confirm that works as expected yet.

As a followup I'm thinking that it may be interesting to consider moving the searchfox page parsing and SYM_INFO usage into its own separate script and create a test that confirms that seems parse successfully what searchfox pages are including (so that we may more easily catch regressions and fix those without any need to debug that code running in the extension content script context).

Let me know how this looks to you (and if there is any further tweaks or fixes needed).

File: phabricator.js [hunk 1]
```diff
@@ -95,23 +95,26 @@ function searchInSearchfox(path, searchfoxDoc) {
...
+  let dataSymbols = searchfoxElem.dataset['symbols'].split(',');
+  for (const dataSymbol of dataSymbols) {
+    let {jumps, sym, pretty} = searchfoxSymInfo[dataSymbol] || {};
 
-    for (let i = 0; i < jumps.length; i++) {
-      let sym = jumps[i].sym;
-      let pretty = jumps[i].pretty;
-      links.push(`<a href="https://searchfox.org/mozilla-central/define?q=${encodeURIComponent(sym)}&redirect=false" target="_blank">Go to definition of ${pretty}</a>`);
+    if (jumps?.def) {
+      links.push(`<a href="https://searchfox.org/mozilla-central/source/${jumps.def}" target="_blank">Go to definition of ${pretty}</a>`);
```
Comment (marco-c) [inline phabricator.js:106]: Are `jumps.def`, `jumps.decl` and `jumps.idl` already URL safe?

**Review by marco-c — APPROVED** [review]
LGTM! @asutherland do you want to take a look too?

**Comment by marco-c** [comment]
> As a followup I'm thinking that it may be interesting to consider moving the searchfox page parsing and SYM_INFO usage into its own separate script and create a test that confirms that seems parse successfully what searchfox pages are including (so that we may more easily catch regressions and fix those without any need to debug that code running in the extension content script context).

Ah yes, that would be ideal!

File: phabricator.js (hunk 1 above)
Comment (rpl) [reply phabricator.js:106]: > Are `jumps.def`, `jumps.decl` and `jumps.idl` already URL safe?

That was my guess, but it is a guess only based on a very quick look to:
-  https://github.com/mozsearch/mozsearch/blob/b31ad1ffd5377e5c83bf15dac1b15a1936ba236a/static/js/context-menu.js#L109
- and https://github.com/mozsearch/mozsearch/blob/b31ad1ffd5377e5c83bf15dac1b15a1936ba236a/static/js/context-menu.js#L253

and so it'd be great to confirm that is actually the case with @asutherland

**Comment by asutherland** [comment]
It's very cool to have a fix for this!  Unfortunately, I no longer am working on searchfox, so it's probably best to direct any questions to https://chat.mozilla.org/#/room/#searchfox:mozilla.org.

**Pull request merged**